lib/media/track/localmediatrack.js

```diff
@@ -24,32 +24,52 @@ function mixinLocalMediaTrack(AudioOrVideoTrack) {
       super(mediaTrackSender, options);
 
       Object.defineProperties(this, {
-        _didCallEnd: {
-          value: false,
-          writable: true
-        },
         _trackSender: {
           value: mediaTrackSender
         },
         id: {
           enumerable: true,
-          value: mediaStreamTrack.id
+          value: mediaTrackSender.id
         },
         isEnabled: {
           enumerable: true,
           get() {
-            return mediaStreamTrack.enabled;
+            return mediaTrackSender.enabled;
           }
         },
         isStopped: {
           enumerable: true,
           get() {
-            return mediaStreamTrack.readyState === 'ended';
+            return mediaTrackSender.readyState === 'ended';
           }
         }
       });
     }
 
+    /**
+     * @private
+     * replaces underlying track and
+     * returns a promise.
+     */
+    _setMediaStreamTrack(mediaStreamTrack) {
+      // Note: ideally we would have liked to change the id
+      // of the new mediaStreamTrack to match exsiting id, but
+      // the property is readOnly and non configurable.
```

Comment:
We should document in CHANGELOG and LocalAudioTrack class that:
* `mediaStreamTrack` can change.
* `started` and `stopped` events can be fired more than once.
* What to do about any event listeners attached to the old MediaStreamTrack by the developer?
```suggestion
```